tests/test_protocols.py

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Oct 21 12:16:49 2022
+
+@author: Alexandra N. Walker
+
+Testing methods from protocols.py
+"""
+from controller.protocols import (
+    connection,
+    didexchange,
+    indy_anoncred_onboard,
+    indy_anoncred_credential_artifacts,
+    indy_issue_credential_v1,
+    indy_issue_credential_v2,
+    indy_present_proof_v1,
+    indy_present_proof_v2,
+    indy_anoncreds_revoke,
+    indy_anoncreds_publish_revocation,
+    indy_anoncreds_proof_revocations
+    )
+from tests.conftest import getenv_or_raise
+from controller.controller import Controller
+import pytest
+import pytest_asyncio
+from os import getenv
+
+@pytest_asyncio.fixture
+async def alice():
+    alice = await Controller(getenv_or_raise("ALICE")).setup()
+    yield alice
+    await alice.shutdown()
+
+@pytest_asyncio.fixture
+async def bob():
+    bob = await Controller(getenv_or_raise("BOB")).setup()
+    yield bob
+    await bob.shutdown()
+
+@pytest.mark.asyncio
+async def test_connection_established(alice,bob):
+    """Testing that connection is established."""
+    alice_conn, bob_conn = await connection(alice, bob)
+    assert alice_conn
+    assert bob_conn
+    yield alice_conn, bob_conn
+    
+@pytest.mark.asyncio
+async def test_did_exchanged(alice,bob):
+    """Testing that dids are exchanged successfully."""
+    alice_conn, bob_conn = await didexchange(alice, bob)
+    assert alice_conn
+    assert bob_conn
+    yield alice_conn, bob_conn
+    
+@pytest_asyncio.fixture
+async def did_exchange(alice,bob):
+    """Testing that dids are exchanged successfully."""
+    alice_conn, bob_conn = await didexchange(alice, bob)
+    assert alice_conn
+    assert bob_conn
+    yield alice_conn, bob_conn
```

Comment:
We should have another test that sets the `multi_use` parameter of `didexchange` to `True` and make sure it works (this was actually broken up until recently).